chiselFrontend/src/main/scala/chisel3/core/Aggregate.scala

```diff
@@ -66,29 +66,32 @@ object Vec {
     // with apply(Seq) after type erasure. Workarounds by either introducing a
     // DummyImplicit or additional type parameter will break some code.
 
-    // Check that types are homogeneous.  Width mismatch for Elements is safe.
-    require(!elts.isEmpty)
-
-    val vec = Wire(new Vec(cloneSupertype(elts, "Vec"), elts.length))
-
-    def doConnect(sink: T, source: T) = {
-      // TODO: this looks bad, and should feel bad. Replace with a better abstraction.
-      // NOTE: Must use elts.head instead of vec.sample_element because vec.sample_element has
-      //       WireBinding which does not have a direction
-      val hasDirectioned = elts.head match {
-        case t: Aggregate => t.flatten.exists(_.dir != Direction.Unspecified)
-        case t: Element => t.dir != Direction.Unspecified
+    if (elts.isEmpty) {
+      // Support empty Vecs.
+      new Vec(null.asInstanceOf[T], 0)
```

Comment:
null.asInstanceOf[T] raises serious red flags.

Possible alternative solutions:
- Make sample_element an Option[T]
- Remove sample_element entirely, using instead head of Vec elements, and handling the empty case properly